actors/builtin/miner/miner_actor.go

```diff
@@ -1610,6 +1611,24 @@ func handleProvingDeadline(rt Runtime) {
 				st.ProvingPeriodStart = st.ProvingPeriodStart + WPoStProvingPeriod
 			}
 		}
+
+		// expire pre-committed sectors
+		expiryQ, err := LoadBitfieldQueue(store, st.PreCommittedSectorsExpiry, quant)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load sector expiry queue")
+
+		bf, modified, err := expiryQ.PopUntil(currEpoch)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to pop expired sectors")
+
+		if modified {
+			st.PreCommittedSectorsExpiry, err = expiryQ.Root()
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to save expiry queue")
+		}
+
+		if bf != nil {
+			depositToBurn, err := st.checkPrecommitExpiry(store, bf)
+			builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to expire pre-committed sectors")
+			burnFunds(rt, depositToBurn)
```

Comment:
Can't call this in a transaction.

Reply:
This should probably just be added to `penaltyTotal` and burnt at the end.

Reply:
This is done.